infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -117,21 +278,15 @@ protected String evalNamespaceName(String workspaceId, Subject currentUser) {
     }
   }
 
-  /**
-   * Creates a Kubernetes namespace for the specified workspace.
-   *
-   * <p>Namespace won't be prepared. This method should be used only in case workspace recovering.
-   *
-   * @param workspaceId identifier of the workspace
-   * @return created namespace
-   */
-  public KubernetesNamespace create(String workspaceId, String namespace) {
-    return doCreateNamespace(workspaceId, namespace);
-  }
-
-  @VisibleForTesting
-  KubernetesNamespace doCreateNamespace(String workspaceId, String name) {
-    return new KubernetesNamespace(clientFactory, name, workspaceId);
+  protected String evalDefaultNamespaceName(String defaultNamespace, Subject currentUser) {
```

Comment:
What would happen if `currentUser` is `Subject.ANONYMOUS`? On hosted Che, we saw anonymous users fairly commonly.

Reply:
Well. I believe it must not happen here.
Do you think it makes sense to throw an exception in such case?

Reply:
If it's not possible, then I'm fine with it as-is. I'm not fully familiar with the subject flow on k8s, etc.